modules/gapi/test/gapi_gpu_test_kernels.cpp

```diff
@@ -0,0 +1,141 @@
+            idxArg = kernel.set(idxArg, (int)size.width);
+            idxArg = kernel.set(idxArg, (int)size.height);
+            idxArg = kernel.set(idxArg, cv::ocl::KernelArg::PtrWriteOnly(out));
+            idxArg = kernel.set(idxArg, (int)out.step);
+            idxArg = kernel.set(idxArg, (int)size.height);
+            idxArg = kernel.set(idxArg, (int)size.width);
+            idxArg = kernel.set(idxArg, (int)tile_y);
+            idxArg = kernel.set(idxArg, cv::ocl::KernelArg::PtrReadOnly(gKer));
+
+            if (!kernel.run(2, globalsize, NULL, false))
+            {
+                printf("symm_7x7_test OpenCL kernel run failed\n");
+            }
+        }
+        else
+        {
+            //printf("symm_7x7 OpenCL kernel run failed - OpenCL is not activated\n");
+
+            //CPU fallback
+            cv::Point anchor = { -1, -1 };
+            double delta = 0;
+
+            int c_int[10] = { 1140, -118, 526, 290, -236, 64, -128, -5, -87, -7 };
+            float c_float[10];
+            for (int i = 0; i < 10; i++)
+            {
+                c_float[i] = c_int[i] / 1024.0f;
+            }
+            // J & I & H & G & H & I & J
+            // I & F & E & D & E & F & I
+            // H & E & C & B & C & E & H
+            // G & D & B & A & B & D & G
+            // H & E & C & B & C & E & H
+            // I & F & E & D & E & F & I
+            // J & I & H & G & H & I & J
+
+            // A & B & C & D & E & F & G & H & I & J
+
+            // 9 & 8 & 7 & 6 & 7 & 8 & 9
+            // 8 & 5 & 4 & 3 & 4 & 5 & 8
+            // 7 & 4 & 2 & 1 & 2 & 4 & 7
+            // 6 & 3 & 1 & 0 & 1 & 3 & 6
+            // 7 & 4 & 2 & 1 & 2 & 4 & 7
+            // 8 & 5 & 4 & 3 & 4 & 5 & 8
+            // 9 & 8 & 7 & 6 & 7 & 8 & 9
+
+            float coefficients[49] =
+            {
+                c_float[9], c_float[8], c_float[7], c_float[6], c_float[7], c_float[8], c_float[9],
+                c_float[8], c_float[5], c_float[4], c_float[3], c_float[4], c_float[5], c_float[8],
+                c_float[7], c_float[4], c_float[2], c_float[1], c_float[2], c_float[4], c_float[7],
+                c_float[6], c_float[3], c_float[1], c_float[0], c_float[1], c_float[3], c_float[6],
+                c_float[7], c_float[4], c_float[2], c_float[1], c_float[2], c_float[4], c_float[7],
+                c_float[8], c_float[5], c_float[4], c_float[3], c_float[4], c_float[5], c_float[8],
+                c_float[9], c_float[8], c_float[7], c_float[6], c_float[7], c_float[8], c_float[9]
+            };
+
+            cv::Mat kernel = cv::Mat(7, 7, CV_32FC1);
+            float* cf = kernel.ptr<float>();
+            for (int i = 0; i < 49; i++)
+            {
+                cf[i] = coefficients[i];
+            }
+
+            cv::filter2D(in, out, CV_8UC1, kernel, anchor, delta, cv::BORDER_REPLICATE);
+        }
+    }
+};
+
+
+cv::gapi::GKernelPackage gpuTestPackage = cv::gapi::kernels
+        <GGPUSymm7x7_test
+        >();
```

Comment:
I also think we might move this stuff to a single file (gapi_gpu_test.cpp) - there's no technical need to have kernels defined in headers, and at the moment it is used in a single place only.

Reply:
Done. My intention was to separate kernel declarations/implementation in separate files to allow reusage in other applications out of the box. Kind of portability. Now everything is in the test source file gapi_gpu_test.cpp.